tools/walletextension/wallet_extension.go

```diff
@@ -142,6 +142,20 @@ func (w *WalletExtension) ProxyEthRequest(request *common.RPCRequest, conn userc
 		}
 	}
 
+	// check if user is sending a new transaction and if we should store it in the database for debugging purposes
+	if request.Method == rpc.SendRawTransaction && w.config.StoreIncomingTxs {
+		userIDBytes, err := common.GetUserIDbyte(hexUserID)
+		if err != nil {
+			w.Logger().Error(fmt.Errorf("error decoding string (%s), %w", hexUserID[2:], err).Error())
+			return nil, errors.New("error decoding userID. It should be in hex format")
+		}
+		err = w.storage.StoreTransaction(request.Params[0].(string), userIDBytes)
```

Comment:
Are we 100% certain that the first param exists, can be cast to a string and will have "0x" as the first two characters so you can strip them off?

Reply:
Things are checked in a function that calls that function, but added additional check there to avoid an edge case that might happen.